code/modules/mob/living/simple_animal/constructs.dm

```diff
@@ -57,7 +57,13 @@
 	if(held_body) // Null check for empty bodies
 		held_body.forceMove(get_turf(src))
 		SSticker.mode.add_cult_immunity(held_body)
-		held_body.key = key
+		if(ismob(held_body)) // Check if the held_body is a mob
+			held_body.key = key
+		else if(istype(held_body, /obj/item/organ/internal/brain)) // Check if the held_body is a brain
+			var/obj/item/organ/internal/brain/brain = held_body
+			if(brain.brainmob) // Check if the brain has a brainmob
+				brain.brainmob.key = key // Set the key to the brainmob
+				brain.brainmob.mind.transfer_to(brain.brainmob) // Transfer the mind to the brainmob
```

Comment:
Because theres so many things here that have the chance to runtime, I think it may be wise to move this section to its own proc, so if it runtimes it doesnt make constructs invincible. It would be nice if the design team could weigh in on this idea

Reply:
Should be done